rust/models/src/status.rs

```diff
@@ -62,6 +62,25 @@ impl Phase {
     }
 }
 
+impl FromStr for Phase {
+    type Err = ();
+
+    fn from_str(status: &str) -> Result<Phase, ()> {
+        match status {
+            "requested" => Ok(Phase::Requested),
+            "new" => Ok(Phase::Requested),
+            "running" => Ok(Phase::Running),
+            "ready" => Ok(Phase::Running),
+            "stopped" => Ok(Phase::Stopped),
+            "failed" => Ok(Phase::Failed),
+            "succeeded" => Ok(Phase::Succeeded),
+            "finished" => Ok(Phase::Succeeded),
+            "stored" => Ok(Phase::Stored),
```

Comment:
I think stored is incorrect here, as we use it to indicate that a scan has been stored but not yet started by the user within the /scan endpoint. In this case the user has started the scan so it should either stay on requested or be Succeeded?

Reply:
Not sure if I understand your point. I use the FromStr trait for converting every possible string in our stack into a valid Phase status. Actually, some strings like `new` (openvas status) and `requested` (openvasd status) corresponds to the same Phase `Requested`.
In this case, `stored` is the an openvasd phase status already defined and will never be set from the openvas side.

Reply:
ah, right, I assumed it is openvas specific as the string cases are but it's an overall fromstr.

Reply:
To avoid confusion, I moved the openvas specif phases to openvas.rs with https://github.com/greenbone/openvas-scanner/pull/1585/commits/ba24c69898fcab5fd5763816533995570388ba91